### PR TITLE
Light Account 2.0

### PR DESCRIPTION
**Pull request by Jon-edge**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

Modify light account creation flow to complete a step sooner - post PIN selection rather than the original TOS scene.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206200642637108

### CODE REVIEW

File: src/hooks/useCreateAccount.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        showError(e)
+        dispatch({
+          type: 'NAVIGATE',
+          data: {
+            name: isLightAccount ? 'newAccountPin' : 'newAccountTos',
+            params: isLightAccount ? {} : createAccountParams
+          }
+        })
+        error = String(e)
+      }
```
Comment (samholmes) [inline src/hooks/useCreateAccount.ts:82]: It would make more sense to put the account type logic at the call-site for this API.

**Review by samholmes — CHANGES_REQUESTED** [review]
We should probably sidebar on this feedback

File: src/components/scenes/ChangePinScene.tsx [hunk 2]
```diff
@@ -186,12 +187,12 @@ export const NewAccountPinScene = (props: SceneProps<'newAccountPin'>) => {
...
+
+  const handleCreateAccount = useCreateAccount(route.params)
+
+  const handleSubmit = useHandler(async (newPin: string) => {
     onLogEvent('Signup_PIN_Valid')
-    dispatch({
-      type: 'NAVIGATE',
-      data: { name: 'newAccountTos', params: { ...route.params, pin: newPin } }
-    })
+    await handleCreateAccount({ ...route.params, pin: newPin })
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/ChangePinScene.tsx:195]: We don't. I also checked with Peter already and he wants the same ending
analytics event for this signup flow, even though we no longer show TOS for
light accounts.

File: src/components/scenes/newAccount/NewAccountTosScene.tsx [hunk 3]
```diff
@@ -196,16 +170,14 @@ export const NewAccountTosScene = (props: NewAccountTosProps) => {
...
+  const { experimentConfig, onLogEvent = () => {} } = useImports()
+  const createAccount = useCreateAccount(route.params)
 
+  const handleNext = useHandler(async () => {
     if (experimentConfig.signupCaptcha === 'withCaptcha') {
       onLogEvent('Signup_Captcha_Shown')
-      const result = await Airship.show<boolean | undefined>(bridge => {
-        return <ChallengeModal bridge={bridge} />
-      })
+      const result = await showChallengeModal()
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/newAccount/NewAccountTosScene.tsx:179]: It was for an older implementation, but I think it's still good to have all
the modals in the same place.

File: src/hooks/useCreateAccount.ts (hunk 1 above)
Comment (Jon-edge) [reply src/hooks/useCreateAccount.ts:82]: Would, but we want the error attached to the `onLogEvent` below

**Review by samholmes — APPROVED** [review]
I applied your fixups, and added a couple of changes as my feedback for you do review and apply after review before publishing.

**Pull request merged**